File: PagingDemo/app/src/main/java/com/example/pagingdemo/UserDao.java
package com.example.pagingdemo;

import androidx.paging.DataSource;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

@Dao
public interface UserDao {

    @Insert
    void insertUser(User... users);

    @Query("DELETE FROM user")
    void deleteUsers();

    @Query("SELECT * FROM user")
    DataSource.Factory<Integer, User> getAllUsers();
}

File: RoomDemo/app/src/main/java/com/example/roomdemo/UserDao.kt
package com.example.roomdemo

import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface UserDao {

    @Query("select * from user")
    fun getAll() : LiveData<List<User>>

    @Delete
    fun deleteUser(user: User)

    @Insert
    fun insertUser(user: User)

    @Insert
    fun insertUsers(vararg user: User)

    @Update
    fun updateUser(user: User)

}
